### ISSUE TITLE
Crash when rendering MultiMeterStatusBar when values are zero

### ISSUE THREAD

**Issue by subpop** [freeorion/freeorion#1272]
### Environment

* **FreeOrion Version**: 0.4.6
* **Operating System**: Linux (Fedora 25)
* **Graphic card used**: Intel HD 530
* **Fetched as**: Compiled from source

### Description
I'm the package maintainer for freeorion in Fedora. I recently received a [crash report](https://bugzilla.redhat.com/show_bug.cgi?id=1416197) that looks like it's caused by an implicit cast from a signed int to an unsigned int (size_t). `length_error` is thrown by std::vector when `reserve` is called and the passed in `size_t` is larger than the size of the vector. That got me to inspect the values as computed inside `MultiMeterStatusBar.cpp` and noticed that when `num_segments` works out to be `0` (as is the case when all the resources of an Outpost on a planet are zero), when `bar_verts.reserve(num_segments - 1)` is called, the expression `num_segments - 1` evaluates to `-1`, but `reserve` implicitly casts this to an unsigned `size_t`. In the debugger, I was able to see these values are huge.

```
(lldb) p bar_verts.b_data.max_size()
(size_type) $5 = 4611686018427387903
(lldb) p (size_type) num_segments - 1
(unsigned long) $6 = 18446744073709551615
```

### Expected Result
The resource status bar to pull down.

### Actual Result
Crash, throwing `length_error`.

### Steps to reproduce
1. Build Outpost ship
2. Deploy the outpost on some planet
3. Click on pull-down icon in outpost detail window ( prod/research/trade/supply where all 4 values are 0 )

**Comment by LGM-Doyle** [comment]
@subpop thank you for the perfect bug report.  Thanks also for packaging freeorion for Fedora.

Thanks to Peter "Pessoft" Kolínek and Link Dupont for additional details on bugzilla.

I've verified this bug.  It is a crash across all platforms, not just linux.

I've provided a fix in #1274.

**Comment by Vezzra** [comment]
We've two different PRs providing a fix for this issue now: #1273 and #1274. #1274 looks more thorough to me, but that's only my opinion. What do the others say? Which one of the fixes should be merged?

@geoffthemedio?

BTW: which status label should be used for the rejected PR? Neither "follow up" nor "invalid" sounds really fitting, any ideas...?

**Comment by LGM-Doyle** [comment]
I should withdraw mine, #1274.  I did not intend to post a second fix to a solved problem.  How about posting it as "duplicate".

**Comment by geoffthemedio** [comment]
Already merged #1274 but my std::max version of #1273 could also be merged for extra clarity / safety if desired.

**Comment by Vezzra** [comment]
> my std::max version of #1273 could also be merged for extra clarity / safety if desired

Sure, can't hurt I guess. Given how small that change is, I think making another PR or rebasing and adjusting #1273 accordingly is overkill - just commit the change you suggested.
